lib/ansible/modules/packaging/os/yum.py

```diff
@@ -793,6 +793,16 @@ def exec_install(module, items, action, pkgs, res, yum_basecmd):
         res['changed'] = False
         module.fail_json(**res)
 
+    # Fail if yum prints 'No space left on device' because that means some
+    # packages failed executing their post install scripts because of lack of
+    # free space (e.g. kernel package couldn't generate initramfs). Note that
+    # yum can still exit with rc=0 even if some post scripts didn't execute
+    # correctly.
+    if 'No space left on device' in (out or err):
+        res['changed'] = False
+        res['error'] = 'No space left on device'
```

Comment:
Actually should this be `msg` as stated in https://github.com/ansible/ansible/pull/40723 too?

Reply:
Yes it should, good catch. +1